app/javascript/app/components/data-explorer-content/data-explorer-filters/data-explorer-filters.js

```diff
@@ -117,14 +108,17 @@ class DataExplorerContentContainer extends PureComponent {
     const { section } = this.props;
     const SOURCE_AND_VERSION_KEY = 'source';
     let paramsToUpdate = [];
-    const dependentKeysToDeleteParams = getDependentKeysToDelete(
-      value,
-      section,
-      filterName
-    ).map(k => ({ name: `${section}-${k}`, value: undefined }));
-    let parsedValue = value;
-    if (multiple) parsedValue = this.parsedMultipleValues(filterName, value);
-    parsedValue = parsedValue === '' ? undefined : parsedValue;
+    const dependentKeysToDeleteParams = DATA_EXPLORER_DEPENDENCIES[section]
+      ? getDependentKeysToDelete(section, filterName).map(key => ({
+        name: `${section}-${key}`,
+        value: undefined
+      }))
+      : [];
+
+    const parsedValue = multiple
+      ? this.parsedMultipleValues(filterName, value)
+      : value;
```

Comment:
Not sure if there is any case in which not forcing empty string (`''`) to `undefined` will trigger any error, that's why I though a function would be needed. But if this is not the case feel free to merge it.
Thanks!! 🙏